helpers/loyverse.py

```diff
@@ -1,133 +1,109 @@
-class LoyverseConnector:
-    BASE_URL = "https://api.loyverse.com/v1.0"
-    READ_ALL_CUSTOMERS_ENDPOINT = f"{BASE_URL}/customers?updated_at_min=2023-07-01T12:30:00.000Z&limit=250"
-    CREATE_OR_UPDATE_CUSTOMER_ENDPOINT = f"{BASE_URL}/customers"
-
-    def __init__(self, token):
-        self.token = token
-
-    @staticmethod
-    def is_number(variable):
-        return isinstance(variable, (int, float))
 
-    @staticmethod
-    def get_customer_from_username(customers, username):
-        customer = customers.get(username)
-        if customer:
-            return customer
-        else:
-            raise Exception(f"@{username} is not a recognised user, try again!")
+def _default(obj):
+    if hasattr(obj, 'to_json'):
+        return obj.to_json()
+    raise TypeError(f'Object of type {obj.__class__.__name__} is not JSON serializable')
 
-    @staticmethod
-    def read_customers(data):
-        customers = {}
 
-        for c in data.get("customers"):
-            note = c.get("note")
-            if note:
-                infos = {
-                    "note": note,
-                    "id": c.get("id"),
-                    "name": c.get("name"),
-                    "total_points": c.get("total_points")
-                }
+class InsufficientFundsError(Exception):
+    pass
 
-                customers[note] = infos
 
-        return customers
+class Customer:
+    def __init__(self, customer_id: str, name: str, username: str, points: Points):
+        self.customer_id = customer_id
+        self.name = name
+        self.username = username
+        self.points = points
 
-    def call_api_get(self, url):
-        headers = {
-            "Authorization": f"Bearer {self.token}"
+    def to_json(self) -> dict:
+        return {
+            'id': self.customer_id,
+            'name': self.name,
+            'note': self.username,
+            'total_points': self.points,
         }
 
-        response = requests.get(url, headers=headers)
-
-        if response.status_code == 200:
-            return response.json()
-        else:
-            print(f"Error {response.status_code} occurred.")
+    @staticmethod
+    def from_json(data: dict) -> Optional["Customer"]:
+        username = data.get("note")
+        if not username:
             return None
 
-    def call_api_post(self, body):
-        headers = {
-            "Authorization": f"Bearer {self.token}",
-            "Content-Type": "application/json"
-        }
+        return Customer(
+            customer_id=data.get("id"),
+            name=data.get("name"),
+            username=username,
+            points=Points(data.get("total_points"))
+        )
 
-        response = requests.post(self.CREATE_OR_UPDATE_CUSTOMER_ENDPOINT, headers=headers, data=json.dumps(body))
 
-        if response.status_code == 200:
-            return response.json()
-        else:
-            print(f"Error {response.status_code} occurred.")
-            return None
+class LoyverseConnector:
```

Comment:
Generally I agree with this.

In this particular instance I thought it was ok because the intention is that the InsufficientFundsError class is just a regular exception by a different name.

Additionally, the Customer class is only meant to be used by LoyverseConnector. This is something that in other languages may be called an "internal" or "protected" class. I'm not aware if Python supports this situation.

In a future pull request, the Customer class will be significantly changed and at that point I will move it into a separate file.